lib/rmt/downloader.rb

```diff
@@ -181,17 +172,38 @@ def finalize_download(request, local_file, checksum_type = nil, checksum_value =
       )
     end
 
-    RMT::ChecksumVerifier.verify_checksum(checksum_type, checksum_value, request.download_path) if (checksum_type && checksum_value)
+    verify_checksum!(checksum_type, checksum_value, request.download_path)
 
     FileUtils.mv(request.download_path.path, local_file)
     File.chmod(0o644, local_file)
 
-    ::RMT::Deduplicator.add_local(local_file, checksum_type, checksum_value) if @save_for_dedup
+    if @track_files && local_file.match?(/\.(rpm|drpm)$/)
+      DownloadedFile.track_file(checksum: checksum_value,
+                                checksum_type: checksum_type,
+                                local_path: local_file,
+                                size: File.size(local_file))
+    end
 
     @logger.info("↓ #{File.basename(local_file)}")
   rescue StandardError => e
     request.download_path.unlink
     raise e
   end
 
+  def verify_checksum!(checksum_type, checksum_value, download_path)
```

Comment:
Shouldn't this method remain in `RMT::ChecksumVerifier` ?

It looks like the reponsability to verify a checksum should be in `ChecksumVerifier`.

Reply:
This method only wraps `RMT::ChecksumVerifier.match_checksum?` to have extra behavior:

1. If there's no checksum information, it just ignores the verification;
1. If the file's checksum doesn't match the expected digest, it raises an exception.

In this PR I refactored the class `RMT::ChecksumVerifier` to change its behavior: previously it had the method `.verify_checksum` which would raise an exception every time a mismatch happened. However, I needed to reuse the verification logic in other places where I didn't need an exception, so I refactored the method to have a boolean return type and decided what to do with it on each class that depends on checksum verifications.

`RMT::Downloader` has its own context: sometimes we download a file (mainly `.xml` repository metadata) and we don't need to verify checksums. If you take a look at the other methods, you'll see some of them receive checksums as optional parameters, with default to `nil`.

Reply:
> This method only wraps `RMT::ChecksumVerifier.match_checksum?` to have extra behavior:
> 
>     1. If there's no checksum information, it just ignores the verification;
> 
>     2. If the file's checksum doesn't match the expected digest, it raises an exception.
> 
> 
> In this PR I refactored the class `RMT::ChecksumVerifier` to change its behavior: previously it had the method `.verify_checksum` which would raise an exception every time a mismatch happened. However, I needed to reuse the verification logic in other places where I didn't need an exception, so I refactored the method to have a boolean return type and decided what to do with it on each class that depends on checksum verifications.

That makes sense to me and I think is a good change and this explanation helps.
> 
> `RMT::Downloader` has its own context: sometimes we download a file (mainly `.xml` repository metadata) and we don't need to verify checksums. If you take a look at the other methods, you'll see some of them receive checksums as optional parameters, with default to `nil`.

Please, correct me if I'm wrong, the method is used once [here](https://github.com/SUSE/rmt/pull/585/files#diff-b5294cd24ec48cddb9d6a240471ec8c4R175) to raise an exception (or not), so maybe a rename to `handle_checksum_verification` (or something more accurate of what the method does) or maybe avoid the method.

Reply:
I renamed the method to `handle_checksum_verification` in the last commit.